docs/developer-docs/best-practices/considerations-for-nft-devs.md

```diff
@@ -0,0 +1,109 @@
+
+You can also use a more complete monitoring solution like [Canistergeek](https://canistergeek.app/). Recently, the team behind Canistergeek added a new feature to their [NFTgeek](https://t5t44-naaaa-aaaah-qcutq-cai.raw.ic0.app/) product that allows observing the [cycles balance of popular NFT collections](https://t5t44-naaaa-aaaah-qcutq-cai.raw.ic0.app/cycles). 
+
+
+
+### Follow best practices for efficient implementations
+
+There are a few foot guns that could make your canister more expensive than you’d expect. Here are a few examples that you might encounter when implementing NFT canisters.
+
+* Use of the heartbeat: A plain heartbeat without doing anything will cost ~0.055 T cycles/day. There are discussions about [implementing alternatives that allow for cheaper scheduling](https://forum.dfinity.org/t/heartbeat-improvements-timers-community-consideration/14201).
+* Some advice for Motoko developers: 
+    * Use `TrieMap` instead of `HashMap` to avoid the performance cliff of automatic resizing associated with HashMaps.
+    * Use `Buffer` instead of `Array` if you need to dynamically resize the structure
+    * Use `Blob` instead of `[Nat8]` for storing large binary assets
+    * Consider using `Blob` instead of `[Nat8]` when sending or receiving Candid `vec nat8/blob` values. The choice is yours but `Blobs are 4x more compact and much less taxing on garbage collection (GC).
+    * Consider storing large `Blob`s in stable memory, to reduce pressure on the GC even further, especially when the manual memory management of that Blob is simple (e.g. they are only added, never deleted).
+    * Consider using the `compacting-gc` setting, especially in append-only scenarios, to allow access to larger heaps and reduce the cost of copying large, stationary objects.
+* Some advice for Rust developers:
+    * Be careful with extensive use of `Vec<u8>` and hence the `String` type if you need to (de-)serialize state for upgrades.
+    * Read [Roman’s blog post on effective Rust canisters](https://mmapped.blog/posts/01-effective-rust-canisters.html)
+
+A general article with [good practices for canister development by Joachim Breitner](https://www.joachim-breitner.de/blog/788-How_to_audit_an_Internet_Computer_canister). 
+
+To make sure you won’t get surprised, you can use the recently added [performance counter API](https://forum.dfinity.org/t/introducing-performance-counter-on-the-internet-computer/14027) to profile your canisters even before going live.
+
+
+### Implement mechanisms to backup and restore state
+
+The IC itself does not yet support backup and restoration of the canister state, but it can be implemented in the canister itself. Regular backups are insurance against the worst-case scenario that a canister gets deallocated or there are issues with upgrading a canister.
```

Comment:
can you point to something that would explain how to do it?

Reply:
Unfortunately, there's no resource on this yet.

Reply:
maybe this? https://forum.dfinity.org/t/backup-restore-function-for-a-canister/12849/3

Reply:
Good find!